src/main/java/com/palantir/refreshable/DefaultRefreshable.java

```diff
@@ -45,20 +47,16 @@
  */
 final class DefaultRefreshable<T> implements SettableRefreshable<T> {
     private static final Logger log = LoggerFactory.getLogger(DefaultRefreshable.class);
-    /**
-     * Every ten times refreshable.map is called without an update we must purge SelfRemovingMapSubscriber instances
-     * whose children have been reaped.
-     */
-    private static final int CLEAN_THRESHOLD = 10;
+    private static final Cleaner REFRESHABLE_CLEANER = Cleaner.create();
 
     private static final int WARN_THRESHOLD = 1000;
 
-    @GuardedBy("this")
-    private final Set<Consumer<? super T>> orderedSubscribers = new LinkedHashSet<>();
+    private final Set<Consumer<? super T>> orderedSubscribers = Collections.synchronizedSet(new LinkedHashSet<>());
```

Comment:
Also, do you even need the synchronized collection?

Reply:
Ah yes you do: it is passed to DefaultDisposable. As you said, updates are infrequent, so probaly not a big deal.

Reply:
Many threads may concurrently subscribe or map under a read lock, and unsubscribe via cleaner or dispose with no lock